pkg/plugin/aggregation/run.go

```diff
@@ -150,18 +150,46 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	}()
 
 	// 4. Launch each plugin, to dispatch workers which submit the results back
-	for _, p := range plugins {
-		cert, err := auth.ClientKeyPair(p.GetName())
-		if err != nil {
-			return errors.Wrapf(err, "couldn't make certificate for plugin %v", p.GetName())
-		}
-		logrus.WithField("plugin", p.GetName()).Info("Running plugin")
-		if err = p.Run(client, cfg.AdvertiseAddress, cert); err != nil {
-			return errors.Wrapf(err, "error running plugin %v", p.GetName())
+	go func() {
+		pluginsRun := map[int]bool{}
```

Comment:
Hi @johnSchnake,

Suggesting the following:

```go
pluginsRun := map[int]struct{}
```

Anonymous (or even empty) structs do not consume any memory in Go. If you're using a map as an addressable flag set, anonymous structs should be used as the value type. Dave Cheney wrote a great piece on this a [few years back](https://dave.cheney.net/2014/03/25/the-empty-struct).

Reply:
@akutz , thanks. Will do. I'm aware of the pattern, I think this is just a bit left over from iterations on the theme.